### PR TITLE
Adding rate matrix index

### PR DESCRIPTION
**Pull request by sonachitchyan**
### :pencil: Description

**Type:**  `infrastructure`

This PR addds the NLTEIndexHelper class which has the output `rate_matrix_index`. This will be used for constructing the rate equation for solving NLTE ionization and NLTE excitation. Index helper returns MultiIndex with atomic number, ion number and whether the species is treated in lte or nlte_ionization. If none of those are the case, it keeps the level number as the third index for further usage in NLTE excitation treatment.

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by wkerzendorf** [comment]
@sonachitchyan can you update the ddescription of the PR

**Comment by wkerzendorf** [comment]
@sonachitchyan also adding tests that just load the infrastructure (not even solving it successfully would be great)

File: tardis/plasma/properties/rate_matrix_index.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                    self.nlte_ionization_species,
+                    nlte_excitation_species,
+                )
+            ),
+            names=levels.names,
+        ).drop_duplicates()
+        return rate_matrix_index
+
+    def calculate_rate_matrix_index(
+        self, levels, nlte_ionization_species, nlte_excitation_species
```
Comment (andrewfullard) [inline tardis/plasma/properties/rate_matrix_index.py:31]: If you already have `self` you should be able to use `self.nlte_ionization_species` in this method

File: tardis/plasma/standard_plasmas.py [hunk 2]
```diff
@@ -28,6 +29,7 @@
     adiabatic_cooling_properties,
     two_photon_properties,
     isotope_properties,
+    nlte_properties_new,
```
Comment (andrewfullard) [inline tardis/plasma/standard_plasmas.py:32]: Why the `_new` in the name?

Comment (sonachitchyan) [reply tardis/plasma/standard_plasmas.py:32]: it was just a temporary solution by me and Christian. We just need a way of differentiating between this and numerical nlte that already exists in the code.

File: tardis/plasma/properties/rate_matrix_index.py (hunk 1 above)
Comment (sonachitchyan) [reply tardis/plasma/properties/rate_matrix_index.py]: done

File: tardis/plasma/standard_plasmas.py (hunk 2 above)
Comment (wkerzendorf) [reply tardis/plasma/standard_plasmas.py:32]: this is confusing. We should find a good workaround

**Comment by wkerzendorf** [comment]
Very nice PR. I think the only thing left is the `_new` thing.

File: tardis/plasma/standard_plasmas.py (hunk 2 above)
Comment (sonachitchyan) [reply tardis/plasma/standard_plasmas.py:32]: I can name it `nlte_solver_properties`

**Review by andrewfullard — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**